### ISSUE TITLE
Collection#each should be called on the #docs array directly

### ISSUE THREAD

**Issue by sni** [asciidoctor/jekyll-asciidoc#35]
Hi,

i am getting this warning with Jekyll 3.0.6
```
Deprecation: Collection#each should be called on the #docs array directly.
                    Called by .../src/_plugins/jekyll-asciidoc.rb:117:in `generate'.
```

I am using the latest jekyll-asciidoc from github.

Thanks

**Comment by mojavelinux** [comment]
It seems like this is going to require a special condition if running with Jekyll 3. However, I'm not quite sure what it is asking. Do you have a reference to where this change was announced in Jekyll so that we understand what needs to change? Maybe this? https://github.com/jekyll/jekyll-redirect-from/issues/75

**Comment by sni** [comment]
Sorry, no idea, just got this warning after upgrading to jekyll 3 and wanted to let you know. It's just
a warning and everything seems to work.

**Comment by Stargator** [comment]
In jekyll/jekyll-redirect-from/issues/75, they seemed to just do a conditional check and skip the test if it's using Jekyll 3 which doesn't fix the deprecation in my opinion. I'm looking at possible solutions for the project I encountered this within.

**Comment by Stargator** [comment]
It seems there have been some changes to site.posts. I recommend reading the following: https://talk.jekyllrb.com/t/what-is-docs-array/1425

**Comment by martin-g** [comment]
I also experience this warning:
```
Deprecation: Collection#reverse should be called on the #docs array directly.
                    Called by /home/martin/git/apache/wicket-site/_plugins/tiered_archives_plugin.rb:25:in `site_payload'.
```
The code for this plugin could be seen at https://github.com/apache/wicket-site/blob/asf-site/_plugins/tiered_archives_plugin.rb#L25
I am not sure whether this plugin is part of Jekyll or someone has added it additionally.

**Comment by Stargator** [comment]
Wicket Site depends on the [Jekyll gem here](https://github.com/apache/wicket-site/blob/asf-site/Gemfile). It relates to the same issue I linked above.

There are some significant changes to the API from Jekyll 2.5.3 to 3.0.0. Great work, @martin-g on the your fix.

**Comment by mojavelinux** [comment]
I think this was resolved when 1.0.1 released (see #36). Therefore, I'm going to close it. If this is still an issue, feel free to reopen.
